### ISSUE TITLE
[Accessibility] [Filters Modal] [Screen Reader] [Label] Phantom "button" is announced on reverse swipe when modal open

### ISSUE THREAD

**Issue by ceciliabelem** [cambiatus/frontend#922]
**Describe the bug**
A phantom "button" component is announced by the screen reader when performing a reverse swipe after opening the "Filters" modal, causing confusion.

**WCAG Criterion Violated**
4.1.2 - Name, Role, Value [A]

**Label**
label

**To Reproduce**
1. Go to Home (logged in)
2. Click on the "Offer" tab
3. Open the "Filters" modal
4. Use a screen reader (e.g., NVDA) and perform a reverse swipe
5. Notice a phantom "button" is announced

**Expected behavior**
No phantom or invisible components should be announced by screen readers.

**Screenshots**
If applicable, add screenshots or screen reader logs.

**Cambiatus Env you are using**
Home > Offer tab > Filters modal

**Desktop (please complete the following information):**
- OS: Windows 10
- Browser: Chrome
- Version: 138.0.7204.101 (64 bits)

**Smartphone (please complete the following information):**
- Device: N/A
- OS: N/A
- Browser: N/A
- Version: N/A

**Additional context**
Tested with NVDA screen reader.
